php-checks/src/test/resources/checks/ImmediatelyReturnedVariableCheck.php

```diff
@@ -12,13 +12,33 @@ function f() {
 }
 
 function h() {
-  $e = new Exception();        // NOK
+  $e = new Exception();        // NOK {{Immediately throw this expression instead of assigning it to the temporary variable "$e".}}
   throw $e;
 }
 
 function l($p) {
 
-  $a =& $p;                    // NOK
+  $a =& $p;                    // NOK {{Immediately return this expression instead of assigning it to the temporary variable "$a".}}
+  return $a;
+}
+
+function l($p) {
+  if (true) {
+    $a = $p;                    // NOK {{Immediately return this expression instead of assigning it to the temporary variable "$a".}}
```

Comment:
@pynicolas I agree that we can skip verifying of invariant messages, but this message contains variable name, so I prefer be insured here.